### PR TITLE
removing the functionality of nvfuser python APIs

### PR DESCRIPTION
**Pull request by jjsjann123**
Removing the functionalities from nvfuser python APIs.

Since the use of nvfuser has been deprecated before the last release cut. We are removing torch script support.

I'll have the next PR to actually remove the code base.

### CODE REVIEW

**Comment by vadimkantorov** [comment]
Maybe put somewhere a note in the docs on why it was deprecated? It's useful because a lot of education materials are published and are not going to be changed, including about nvfuser. So it's good to have some trace of its prior existence in the docs, even if just saying that it was available in these versions and was deprecated for tehse reasons.

**Comment by jjsjann123** [comment]
failing test doesn't seem to be relevant.
```
  File "inductor/test_torchinductor_codegen_dynamic_shapes.py", line 332 in <module>
FINISHED PRINTING LOG FILE of inductor/test_torchinductor_codegen_dynamic_shapes 1/4 (/var/lib/jenkins/pytorch/test/test-reports/inductor-test_torchinductor_codegen_dynamic_shapes_cy7zi9za.log)

Running inductor/test_torchinductor_dynamic_shapes 3/4 ... [2023-09-27 02:36:22.180530]
inductor/test_torchinductor_codegen_dynamic_shapes 1/4 failed! Received signal: SIGIOT
Executing ['/opt/conda/envs/py_3.8/bin/python', '-bb', 'inductor/test_torchinductor_dynamic_shapes.py', '--shard-id=2', '--num-shards=4', '-v', '-vv', '-rfEX', '-p', 'no:xdist', '--use-pytest', '--sc=inductor/test_torchinductor_dynamic_shapes_2', '-x', '--reruns=2', '--import-slow-tests', '--import-disabled-tests'] ... [2023-09-27 02:36:22.181133]
/var/lib/jenkins/pytorch/tools/stats/upload_metrics.py:163: UserWarning: Error uploading metric td_experiment_1 to DynamoDB: Unable to locate credentials
  warn(f"Error uploading metric {metric_name} to DynamoDB: {e}")
```

**Comment by jjsjann123** [comment]
> Maybe put somewhere a note in the docs on why it was deprecated? It's useful because a lot of education materials are published and are not going to be changed, including about nvfuser. So it's good to have some trace of its prior existence in the docs, even if just saying that it was available in these versions and was deprecated for these reasons.

We do have release note for that.
2.1 has nvfuser in deprecation for TorchScript. Removing it is the natural follow-up step for that.

I'm open-minded if there's a better place to put a note for that.

File: torch/csrc/jit/OVERVIEW.md [hunk 1]
```diff
@@ -1198,8 +1197,6 @@ or switching the fuser could also provide a temporary fix in case of bugs.
 | NNC on CPU | `torch._C._jit_override_can_fuse_on_cpu()` |
 | NNC on GPU | `torch._C._jit_override_can_fuse_on_gpu()` |
 | NNC context manager | `with torch.jit.fuser("fuser1"):` |
-| NVFuser enable/disable | `torch._C._jit_set_nvfuser_enabled()` |
```
Comment (davidberard98) [inline torch/csrc/jit/OVERVIEW.md:1201]: Can we leave these comments, maybe with a note about deprecation?

File: torch/csrc/jit/OVERVIEW.md [hunk 2]
```diff
@@ -1210,7 +1207,6 @@ or switching the fuser could also provide a temporary fix in case of bugs.
 | NNC enable/disable | `torch::jit::setTensorExprFuserEnabled(bool);` | [here](https://github.com/pytorch/pytorch/blob/1a7e560adecb0192f69f4d05b990800b60dc380b/torch/csrc/jit/passes/tensorexpr_fuser.h#L22) |
 | NNC on CPU | `torch::jit::overrideCanFuseOnCPU(bool);` | [here](https://github.com/pytorch/pytorch/blob/1a7e560adecb0192f69f4d05b990800b60dc380b/torch/csrc/jit/codegen/fuser/interface.h#L28-L29) |
 | NNC on GPU | `torch::jit::overrideCanFuseOnGPU(bool);` | [here](https://github.com/pytorch/pytorch/blob/1a7e560adecb0192f69f4d05b990800b60dc380b/torch/csrc/jit/codegen/fuser/interface.h#L28-L29) |
-| NVFuser enable/disable | `torch::jit::fuser::cuda::setEnabled(bool);` | [here](https://github.com/pytorch/pytorch/blob/1a7e560adecb0192f69f4d05b990800b60dc380b/torch/csrc/jit/codegen/cuda/interface.h#L56) |
```
Comment (davidberard98) [inline torch/csrc/jit/OVERVIEW.md:1213]: and these comments too

**Review by davidberard98 — APPROVED** [review]
LGTM but can you leave a few of the comments in OVERVIEW.md?

**Pull request closed without merging**